### PR TITLE
docs: clarify the purpose of --bin

### LINKED ISSUE #1677: 🛠 Tooling: can --bin be used to create CLI tools?
### Bug Report Checklist

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have pulled the latest `main` branch of the repository.
- [X] I have [searched for related issues](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aissue) and found none that matched my issue.

### Overview

I write CLI tools with TypeScript.  I've started trying to use CTA for this and realised that either the `bin` functionality doesn't create all the `src/bin` code that CTA itself has built in.  I'm assuming this is deliberate?

Either way, I'm just about to (for the second time) copy paste `src/bin` code into a project I'm starting as it's so gosh darn useful!  

Does CTA create this stuff if you just do it the right way?  I can see a `--bin` option here:

https://github.com/JoshuaKGoldberg/create-typescript-app/blob/main/docs/Options.md#optional-options

But it doesn't appear to do anything more than `package.json` fiddling.  Am I getting this right?

### Additional Info

-

### PR DESCRIPTION
**Pull request by johnnyreilly**
## PR Checklist

- [x] Addresses an existing open issue: fixes #1677
- [x] That issue was marked as [`status: accepting prs`](https://github.com/JoshuaKGoldberg/create-typescript-app/issues?q=is%3Aopen+is%3Aissue+label%3A%22status%3A+accepting+prs%22)
- [x] Steps in [CONTRIBUTING.md](https://github.com/JoshuaKGoldberg/create-typescript-app/blob/main/.github/CONTRIBUTING.md) were taken

## Overview

Clarify what `bin` does and does not do.

### CODE REVIEW

File: docs/Options.md [hunk 1]
```diff
@@ -53,7 +53,7 @@ The setup scripts also allow for optional overrides of the following inputs whos
 
 - `--access` _(`"public" | "restricted"`)_: Which [`npm publish --access`](https://docs.npmjs.com/cli/commands/npm-publish#access) to release npm packages with (by default, `"public"`)
 - `--author` _(`string`)_: Username on npm to publish packages under (by default, an existing npm author, or the currently logged in npm user, or `owner.toLowerCase()`)
-- `--bin` _(`string`)_: `package.json` bin value to include for npx-style running.
+- `--bin` _(`string`)_: `package.json` bin value to include for npx-style running. (eg `"./bin/index.js"`) - this does not create any of folders or files; just references to them in the `package.json` bin property
```
Comment (JoshuaKGoldberg) [inline docs/Options.md:56]: 🤔 This is a lot of text for a list item, and still doesn't really cover the nuance or recommended usage here. How about:

* Trimming this down a bit to more precisely just describe it as a `package.json` bin property string
* Writing an FAQ with more information & suggested way to do it

? That way we get the idea of _"export a `bin()`"_ suggested to users, without forcing them to do anything.

```suggestion
- `--bin` _(`string`)_: value to set in `package.json`'s `"bin"` property, per [FAQs > How can I use `bin`?]( ... )
```

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
A start! WDYT?

**Comment by johnnyreilly** [comment]
There - any good?

File: docs/FAQs.md [hunk 2]
```diff
@@ -218,3 +218,11 @@ If you know of any accessibility organization that's published more formal recom
...
 
 If you really want spaces in your project you can always remove the `"useTabs": true`.
+
+## How can I use `bin`?
+
+The `--bin` option allows you to create a `package.json` bin value to include for npx-style running.
+An example of this would be `"./bin/index.js"`.
+You'll need to create the folders and files that `bin` references.
+
+If you'd like an example of what that looks like, just take a look at the [CTA source code](https://github.com/JoshuaKGoldberg/create-typescript-app/blob/e7fafcb8968f8f6c551ab0917c9a6a849a3cba28/bin/index.js)!
```
Comment (JoshuaKGoldberg) [inline docs/FAQs.md:228]:
```suggestion
If you'd like an example of what that looks like, take a look at the [CTA source code](https://github.com/JoshuaKGoldberg/create-typescript-app/blob/e7fafcb8968f8f6c551ab0917c9a6a849a3cba28/bin/index.js)!
```

https://tipsforteachers.substack.com/p/59-stop-saying-the-word-just?open=false#%C2%A7a-tip-to-try-in-class-this-week

**Review by JoshuaKGoldberg — APPROVED** [review]
Great, thanks! 🙌 

Just one nit that I can apply and then merge. Yay!

**Comment by johnnyreilly** [comment]
@JoshuaKGoldberg this is .... un-just 😉

**Pull request merged**